Utilities/OpenCorePackage/Script/Legacy/MountESP

```diff
@@ -0,0 +1,130 @@
+        # with the partition uuid we can find the real disk in in diskutil list output
+        if [[ -n "$UUID" ]]; then
+            realDisk=$(LC_ALL=C diskutil list | \
+                grep -B 1 "$UUID" | \
+                grep -i 'logical volume' | awk '{print $4}' | \
+                sed -e 's/,//g' | sed -e 's/ //g')
+            if [[ -n "$realDisk" ]]; then
+                DiskDevice=$(LC_ALL=C diskutil info "${realDisk}" 2>/dev/null | \
+                                             sed -n 's/.*Part [oO]f Whole: *//p')
+            fi
+        fi
+    fi
+fi
+
+# check if target volume is APFS, and therefore part of an APFS container
+if [[ "$(echo $(LC_ALL=C diskutil list "$DiskDevice" | grep -i 'APFS Container Scheme' | \
+          awk '{print tolower($0)}'))" == *"apfs container scheme"* ]]; then
+    # ok, this disk is an APFS partition, extract physical store device
+    realDisk=$(LC_ALL=C diskutil list "$DiskDevice" 2>/dev/null | \
+        sed -n 's/.*Physical Store *//p')
+    echo Target volume "$1" on "$DiskDevice" is APFS on physical store "$realDisk"
+    DiskDevice=$(LC_ALL=C diskutil info "$realDisk" 2>/dev/null | \
+        sed -n 's/.*Part [oO]f Whole: *//p')
+fi
+
+# check one more time for RAID.
+if [[ ! -z "$(echo $(LC_ALL=C diskutil ar list | grep "$DiskDevice$"))" ]]; then
+  realDisk="$(diskutil ar list | grep -A 10 "${DiskDevice}$" | sed -n 's/Online//p' | head -1 | sed 's/[0-9]*[ ]*\([^ ]*\)[ ]*.*$/\1/g')"
+  # Find first online disk.
+  echo Target volume "$1" on "$DiskDevice" is RAID on physical store "$realDisk"
+  DiskDevice=$(LC_ALL=C diskutil info "$realDisk" 2>/dev/null | \
+  sed -n 's/.*Part [oO]f Whole: *//p')
+fi
+
+
+# echo "realDisk = $realDisk"
+# echo "UUID = $UUID"
+# echo "DiskDevice = $DiskDevice"
+
+# Check if the disk is a GPT disk
+disk_partition_scheme=$("$partutil" --show-partitionscheme "$DiskDevice")
+
+if [[ "$disk_partition_scheme" = "GUID_partition_scheme" ]]; then
+
+    plistbuddy='/usr/libexec/PlistBuddy'
+
+    # Get the index of the ESP device
+    index=$(LC_ALL=C /usr/sbin/gpt -r show "/dev/$DiskDevice" 2>/dev/null | \
+     awk 'toupper($7) == "C12A7328-F81F-11D2-BA4B-00A0C93EC93B" {print $3; exit}')
+    [[ -z "$index" ]] && index=1 # if not found use the index 1
+
+
+
+    if [[ -e /useespfinder ]]; then
+        echo "using espfinder" > /useespfinder
+        ESPDevice=$("$espfinder" -t "${DEST_VOL}")
+        if [[ "$ESPDevice" != disk* ]]; then
+            ESPDevice="${DiskDevice}s${index}"
+        fi
+    else
+        # Define the ESPDevice
+        ESPDevice="${DiskDevice}s${index}"
+    fi
+
+    # Get the ESP mount point if the partition is currently mounted
+    ESPMountPoint=$("$partutil" --show-mountpoint "$ESPDevice")
+
+    if [[ -n "$ESPMountPoint" ]]; then
+        # If already mounted it's okay
+        exitcode=0
+    else
+        # Else try to mount the ESP partition
+        ESPMountPoint="/Volumes/ESP"
```

Comment:
If it exists, maybe try to bruteforce a new name (either adding an incrementing suffix or using a reasonably long randomised suffix)?